Q-9. Given the score of students in multiple exams
Test the hypothesis that the mean scores of all the students are the same. If not, name the student with the highest score.


In [1]:
import pandas as pd
import numpy as np

In [2]:
data = {"Name" : ['karan','deepa','kartik','chandan','jeevan'],
"Exam1" : [85,70,90,75,95],
"Exam2" : [90,80,85,70,92],
"Final_exam" : [92,85,88,75,96]}


In [3]:
df = pd.DataFrame(data)
df

Name  Exam1  Exam2  Final_exam
0    karan     85     90          92
1    deepa     70     80          85
2   kartik     90     85          88
3  chandan     75     70          75
4   jeevan     95     92          96

In [5]:
mean_lst = []
for i in range(df.shape[0]):
    mean = np.mean(df.loc[i][1:])
    name = df.loc[i][0]
    mean_lst.append((name,round(mean,2)))

In [7]:
df2 = pd.DataFrame(mean_lst,columns=['Name',"Percentage"])

In [8]:
df2

Name  Percentage
0    karan       89.00
1    deepa       78.33
2   kartik       87.67
3  chandan       73.33
4   jeevan       94.33

In [9]:
df = df.merge(df2)

In [10]:
df.head()

Name  Exam1  Exam2  Final_exam  Percentage
0    karan     85     90          92       89.00
1    deepa     70     80          85       78.33
2   kartik     90     85          88       87.67
3  chandan     75     70          75       73.33
4   jeevan     95     92          96       94.33

In [11]:
df[df['Percentage'] == df['Percentage'].max()]

Name  Exam1  Exam2  Final_exam  Percentage
4  jeevan     95     92          96       94.33

We can see from above that mean of every student is different which is relatively higher,There is no sufficient data to proof that mean scores of all the students are the same.
where as student with the highest mean score is 

 Jeevan who has scored 94.33 %
